testcases/OpTestFastReboot.py

```diff
@@ -102,8 +105,10 @@ def runTest(self):
             self.skipTest("Qemu platform doesn't have fast-reboot support")
 
         if self.boot_to_os():
+            self.cv_SYSTEM.sys_set_bootdev_no_override()
             self.cv_SYSTEM.goto_state(OpSystemState.OS)
         else:
+            self.cv_SYSTEM.sys_set_bootdev_setup()
```

Comment:
Shouldn't these be set by the state machine doing the goto_state() processing?

Or, rather, I guess the question is "What am I missing here?"

Reply:
> set by the state machine

The cases observed were when a box was left with who-knows-what setting and then when the detection grabbed the box (so we just picked up the shell/OS) and later when we reboot we do not know how the box was setup which can lead to unexpected results.